### DISCUSSION TITLE
websockets ssl node.js

### DISCUSSION THREAD

**Discussion by GREEB** [perspective-dev/perspective#1894] [Q&A] [answered]
Hello guys first thanks a lot for perspective amazing library.

I've build a app using perspective and nextjs with web-sockets, in the process of deployment i had some ssl problems, what is the best way to wrap the WebSocketServer into a HTTPS server i would like to avoid forking. 

Thanks a lot for any help

**Comment by texodus** [comment, marked as answer]
The `@finos/perspective` module exports the `WebSocketManager` class, which is a framework-agnostic state manager for multi-client servers.  The [`/examples/react-express` example](https://github.com/finos/perspective/blob/master/examples/remote-express/src/server.ts) uses the class to server Perspective Tables from `express.js` server.  You can also see how Perspective itself implements this in the [implementation of `WebSocketServer`](https://github.com/finos/perspective/blob/master/packages/perspective/src/js/perspective.node.js#L198).

**Reply by GREEB** [reply]
thank you very much
